tests/integration/v3_auth_test.go

```diff
@@ -47,6 +47,33 @@ func TestV3AuthEmptyUserGet(t *testing.T) {
 	}
 }
 
+// TestV3AuthEmptyUserPut ensures that a put with an empty user will return an empty user error,
+// and the consistent_index should be moved forward even the apply-->Put fails.
+func TestV3AuthEmptyUserPut(t *testing.T) {
+	integration.BeforeTest(t)
+	clus := integration.NewCluster(t, &integration.ClusterConfig{
+		Size:          1,
+		SnapshotCount: 3,
+	})
+	defer clus.Terminate(t)
+
+	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	defer cancel()
+
+	api := integration.ToGRPC(clus.Client(0))
+	authSetupRoot(t, api.Auth)
+
+	// The SnapshotCount is 3, so there must be at least 3 new snapshot files being created.
+	// The VERIFY logic will check whether the consistent_index >= last snapshot index on
```

Comment:
Not sure if tests should be dependent on the external behavior. I possible I would prefer to test do the verification `CI > last snapshot index` itself as it is what we are testing here.

Reply:
Yes, it's a good point. But we can only perform the verification after the member has already closed, because the db can only be opened by one user for now. It's exactly the reason why we do the verification on cluster terminating.